sdk/metricsadvisor/ai-metrics-advisor-rest/src/metricsAdvisorClient.ts

```diff
@@ -0,0 +1,144 @@
+import { ClientOptions } from "@azure-rest/core-client";
+import { isTokenCredential, TokenCredential } from "@azure/core-auth";
+import { OperationOptions } from "@azure/core-client";
+import { PagedAsyncIterableIterator, PageSettings } from "@azure/core-paging";
+import { GeneratedClientLike } from "./generated/clientDefinitions";
+import GeneratedClient from "./generated/generatedClient";
+import { AlertingResultQuery } from "./generated/models";
+import { AlertResultOutput } from "./generated/outputModels";
+import { AlertQueryTimeMode, AlertsPageResponse, AnomalyAlert } from "./models";
+import { paginate } from "./paginateHelper";
+import {
+  createMetricsAdvisorKeyCredentialPolicy,
+  MetricsAdvisorKeyCredential,
+  X_API_KEY_HEADER_NAME,
+} from "./metricsAdvisorKeyCredentialPolicy";
+/**
+ * Options for listing alerts
+ */
+export interface ListAlertsOptions extends OperationOptions {
+  /** Number of items to skip */
+  skip?: number;
+}
+
+export class MetricsAdvisorClient {
+  /**
+   * Url to service endpoint
+   */
+  public readonly endpointUrl: string;
+
+  /**
+   * A reference to the auto-generated MetricsAdvisor HTTP client.
+   */
+  private readonly client: GeneratedClientLike;
+
+  constructor(
+    endpointUrl: string,
+    credential: TokenCredential | MetricsAdvisorKeyCredential,
+    options: ClientOptions = {}
+  ) {
+    this.endpointUrl = endpointUrl;
+    this.client = GeneratedClient(endpointUrl, credential, {
+      ...options,
+      credentials: {
+        scopes: ["https://cognitiveservices.azure.com/.default"],
+        apiKeyHeaderName: X_API_KEY_HEADER_NAME
+      }
+    });
+    if (!isTokenCredential(credential)) {
+      this.client.pipeline.addPolicy(createMetricsAdvisorKeyCredentialPolicy(credential));
+    }
+  }
+
+  public listAlerts(
+    alertConfigId: string,
```

Comment:
@joheredi I implemented the same [function](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/metricsadvisor/ai-metrics-advisor/src/metricsAdvisorClient.ts#L382) but igonring the byPage func. Please help advise whether we could leverage helper func in `paginateHelper.ts` or azure core to implement it?